internal/controllers/insights/test/resources.go

```diff
@@ -118,6 +120,12 @@ func (r *InsightsTestResources) NewInsightsProxySecret() *corev1.Secret {
 								"header": "Authorization",
 								"value_type": "plain",
 								"value": "Bearer world"
+							  },
+							  {
+								"op": "set",
+								"header": "User-Agent",
+								"value_type": "plain",
+								"value": "cryostat-operator/%s cluster/abcde"
```

Comment:
Why doesn't the `%s` format specifier for the operator version get replaced here?

Reply:
I wasn't sure about this. Ideally we would hardcode the version here, but that would add an additional step to update every time we alter the version. Also if any developers override `IMAGE_VERSION` in the Makefile for their own uses, it would cause the test to fail. What do you think?

Reply:
Hmm, I see. If there is no straightforward way to inject the actual build version number into the test harness so that the test compares the real expected value to what the controller generates, then I think leaving it like this is okay.

Reply:
We could decouple the version used to generate this constant from the version used as an image tag. and then hardcode the version in the test. This would prevent the latter issue where developers might want to . It will still need manual updating when we bump versions, but maybe that's worth it. How does that sound?

Reply:
If that isn't too much work then it sounds like a reasonable middle ground. At least that way there is a test that validates that this resources normally gets generated properly, and it isn't too annoying in normal development workflows. If it's just a string to bump at release time that's pretty good.